### ISSUE TITLE
Edge doesnt care about partially filled orders

### ISSUE THREAD

**Issue by Pascal66** [freqtrade/freqtrade#4303]
## Describe your environment

  * Operating system: Ubuntu
  * Python Version: last
  * CCXT version: last
  * Freqtrade Version: develop-last
  
## Describe the problem:

*Edge doesnt care about partially filled orders*
`capital in trade` and `available capital` are wrongly calculated 203 & 255 respectivly
I've a capital of ~100USDT
I've an old running partialy filled order(DOGE/USDT) ~95%
I've a running order(XVG/USD)

### Steps to reproduce:

  1. Enable edge
  2. Partially fill an order
  ```
2021-02-03 09:17:33,686 - freqtrade.wallets - INFO - Wallets synced.
2021-02-03 09:17:33,688 - freqtrade.edge.edge_positioning - INFO - winrate: 0.15384615384615385, expectancy: 0.04938862928316878, position size: 1.94584966, pair: BTC/USDT, capital in trade: 203.0676097018202, free capital: 1.94584966, total capital: 54.7203447, stoploss: -0.03, available capital: 255.21007485780203.
2021-02-03 09:17:33,689 - freqtrade.freqtradebot - WARNING - Unable to create trade for BTC/USDT: Available balance (-0.10428493361820301 USDT) is lower than stake amount (0 USDT)
2021-02-03 09:17:35,084 - freqtrade.freqtradebot - INFO - Found open order for Trade(id=44, pair=DOGE/USDT, amount=3153.06928973, open_rate=0.03278409, open_since=2021-02-01 22:45:07)
2021-02-03 09:17:36,581 - freqtrade.wallets - INFO - Wallets synced.
2021-02-03 09:17:38,075 - freqtrade.freqtradebot - INFO - Found open order for Trade(id=47, pair=XVG/USDT, amount=3231.17912155, open_rate=0.01630029, open_since=2021-02-03 08:15:15)
2021-02-03 09:17:41,074 - freqtrade.wallets - INFO - Wallets synced.
```

**Comment by zn3x** [comment]
What do you expect edge doing of partially filled orders? Can you describe what you want vs what actually happens.
The edge positioning is per pair, so the only thing we see the one for btc/usdt pair

**Comment by Pascal66** [comment]
I just expect that the part : 
`capital in trade: 203.0676097018202, free capital: 1.94584966, total capital: 54.7203447, stoploss: -0.03, available capital: 255.21007485780203.` say something true

**Comment by xmatthias** [comment]
from your issue description, it's not clear what would be correct - or even what would be wrong or why it's wrong.

On the one hand, you say you have 100$ Capital - is this the total capital? what's still available?
You've also got a DOGE order partially filled - but with edge, it's unclear what 95% filled means, as it could be an order for a total of 20$, or an order for a total of 100$.

What we'll need to start thinking about what could be wrong:
* Correct amounts of DOGE and USD
* Trade size for all open orders (stake amount)

**Comment by Pascal66** [comment]
I've 100$ total capital
I've a partially filled DOGE/USDT of 95%
I've another trade XVG/USDT with that 95$
So I still have 100$ (5 for DOGE/USDT pending & 95 for XVG/UDST)

Then 
my capital in trade is 95(XVG/USDT) + 5(DOGE/USDT)
my free capital is 0 more or less depending of something else, 0.99 * 100 in my case

**Comment by xmatthias** [comment]
> I've a partially filled DOGE/USDT of 95%

when you say 95% filled - is the order still open on the exchange, or has it been cancelled on the exchange?
Same with XVG obviously...

> free capital: 1.94584966, total capital: 54.7203447

This is taken directly from the balance endpoint, which tells me that you have 1.9$ available to trade - and 54$ tied up in buy orders (so if DOGE is the only open buy-order, it would still have ~53$ outstanding).

the "capital in trade" part is absolutely unclear by the information you provided (what it should be, ...). - best post the output of `/status`  so it's clear which trades are open how.

Based on the log output, you'll have a total of 203 in open trades (trades that are marked with "is_open = True").

----

Your total capital does not really match with the remaining info.

if your TOTAL capital is 100$ - and you have 95$ in XVG (which you don't mention as being an unfilled order), then DOGE would be a 5$ trade, out of which 95% is filled (if i understood this correctly).

The output in the logs indicates that you have ~53$ in open buy-orders (that's the difference between total capital and free capital), so best check which orders are still open and which aren't (these can also be non-bot orders), as the data suggests that the XVG order is also unfilled.

**Comment by Pascal66** [comment]
I just hope you are not kidding me.
I havent : 
`capital in trade: 203.0676097018202, free capital: 1.94584966, total capital: 54.7203447, stoploss: -0.03, available capital: 255.21007485780203`
This is false.
I have : 
`capital in trade: 103.0676097018202, free capital: 1.94584966, total capital: 4.7203447, stoploss: -0.03, available capital: 5.21007485780203`
Then no strategies can work. Because calculation are false.

**Comment by xmatthias** [comment]
All i'm doing is reading your (incomplete) information, and trying to piece together what your actual position / trades are.

your log says the following (see first blog post).

> `2021-02-03 09:17:33,688 - freqtrade.edge.edge_positioning - INFO - winrate: 0.15384615384615385, expectancy: 
0.04938862928316878, position size: 1.94584966, pair: BTC/USDT, capital in trade: 203.0676097018202, free capital: 1.94584966, total capital: 54.7203447, stoploss: -0.03, available capital: 255.21007485780203.`

As far as i can tell for now, the log is not wrong (unless you use a modified version of freqtrade), but the information you are giving us are.
Since it's open source, you can trace the information flow (a simple example for "capital in trade"):

It's passed to edge's `stake_amount()`" (the function generating this log) as 4th parameter follows:

https://github.com/freqtrade/freqtrade/blob/caa3e1a7fa655302618d4113b6f506f337a54639/freqtrade/freqtradebot.py#L452-L458

The 4th parameter is `Trade.total_open_trades_stakes()` - which is defined as follows:

https://github.com/freqtrade/freqtrade/blob/caa3e1a7fa655302618d4113b6f506f337a54639/freqtrade/persistence/models.py#L657-L665

This will result in the following query ran on the database:

``` sql
select sum(stake_amount) from trades
where is_open = true;
```

There is 0 modification to this parameter between quering the database and posting it to the log.
Therefore, by reading your log - you do have 203 USDT currently in Trades.

You may THINK this is not the case, but the log, and the database will not agree with you.

Unfortunately, the information you provide is quite incomplete - so i have to TRY and piece together what your full bot's situation is.
